src/__tests__/tasks/FindTranslation.test.js

```diff
@@ -0,0 +1,234 @@
+
+    wrapper.setState({ currentWord: wordsForTask.length - 1, })
+    instance.toNextWord(true)
+    expect(wrapper.exists('ResultPopup')).toBeTruthy()
+    expect(wrapper.state().answersResult).toEqual([true, false, true])
+  })
+})
+
+describe('check ResultPopup component', () => {
+  const goToTasks = jest.fn()
+  const props = { result: [true, false, true, true, false], goToTasks, }
+
+  let wrapper
+  beforeEach(() => {
+    wrapper = shallow(<ResultPopup {...props} />)
+  })
+
+  test('should finish task with correct task results', () => {
+    const rightAnswersCount = 3
+    wrapper.find('TouchableButton').simulate('press')
+
+    expect(goToTasks).toHaveBeenCalledWith(
+      props.result.length,
+      rightAnswersCount
+    )
+  })
+})
+
+describe('check ProgressBar component', () => {
+  const props = { countWords: 5, answersResults: [true, false, true], }
+
+  let wrapper
+  let instance
+  beforeEach(() => {
+    wrapper = shallow(<ProgressBar {...props} />)
+    instance = wrapper.instance()
+  })
+
+  test('should generate progress parts', () => {
+    const generatePartsSpy = jest.spyOn(instance, 'generateProgressParts')
+    instance.forceUpdate()
+
+    expect(generatePartsSpy).toHaveBeenCalledTimes(1)
+    expect(generatePartsSpy).toHaveBeenCalledWith(props.answersResults)
+  })
+})
+
+describe('check Connector component', () => {
+  const toNextWord = jest.fn()
+  const props = {
+    defaultResult: true,
+    toNextWord,
+    possibleAnswers: ['всемогущий', 'рыба', 'дом', 'собака'],
+    currentWord: {
+      id: 2,
+      word: 'dog',
+      transcription: 'dɒɡ',
+      translation: ['собака'],
+      url: 'https://www.guidedogs.org/wp-content/uploads/2018/01/Mobile.jpg',
+      examples: ['He dogged her every move.'],
+      tagName: 'secondTag',
+    },
+    countWordsForTask: 5,
+    answersResult: [],
+  }
+
+  let wrapper
+  let instance
+  beforeEach(() => {
+    wrapper = shallow(<Connector {...props} />)
+    instance = wrapper.instance()
+  })
+
+  test('should initialize the state correctly', () => {
+    expect(wrapper.state().currentResult).toEqual(props.defaultResult)
+  })
+
+  test('should generate answers buttons', () => {
+    const spy = jest.spyOn(instance, 'generateAnswersButtons')
+    instance.forceUpdate()
+
+    expect(spy).toHaveBeenCalledWith(
+      props.possibleAnswers,
+      props.currentWord.translation
+    )
+  })
+
+  test('should check an answer result correctly', () => {
+    instance.checkResult(true)
+
+    expect(toNextWord).toHaveBeenCalledWith(true)
+    toNextWord.mockClear()
+
+    instance.checkResult(false)
+    expect(toNextWord).not.toHaveBeenCalled()
+    expect(wrapper.state().currentResult).toBeFalsy()
+    toNextWord.mockClear()
+
+    instance.checkResult(true)
+
+    expect(toNextWord).toHaveBeenCalledWith(false)
+    expect(wrapper.state().currentResult).toBeTruthy()
+  })
+})
+
+describe('check SelectableAnswer component', () => {
```

Comment:
Why all these components tests in one file?

Reply:
Because these small components depend only on FindTranslation component, and other components don't know about these components. A want to use the FandTranslation like an independent module